### PR TITLE
feat(ai): expose custom summary prompts in the UI

### LINKED ISSUE #199: [Feature]: Custom summary prompts are unreachable in the UI (backend already supports them)
## Problem / Use Case

Custom **summary** prompts appear to be a half-wired feature: the backend and data model fully support them, but there is no UI path to create one or select it, so the capability is unreachable for self-hosters.

Concretely, I want a single custom summary prompt (e.g. one that detects recording type and formats accordingly) and have it used as my default for new summaries and on per-recording re-generate. Today there is no way to do this through the UI — only the 4 built-in presets are ever selectable, and the title-generation prompt manager (Settings → Providers → Prompts) is the *only* place custom prompts can be created (and those are title prompts, not summary prompts).

## Evidence (against `main` @ `33f0270`)

The backend already resolves custom summary prompts:
- `src/lib/ai/summary-presets.ts` — `CustomSummaryPrompt`, `SummaryPromptConfiguration { selectedPrompt, customPrompts }`, `getAllSummaryPrompts()`, and `getSummaryPromptById()` already merge presets + customs.
- `src/app/api/recordings/[id]/summary/route.ts:96` — `const selectedPreset = presetId || promptConfig.selectedPrompt || "general"` resolves either a preset **or** a custom id.
- `src/app/api/settings/user/route.ts` — `PUT` stores the whole `summaryPrompt` config verbatim (encrypted); `GET` decrypts it. Custom prompts round-trip fine.

But three frontend surfaces hard-code presets and never expose customs:

1. **Settings → Summary** (`src/components/settings-sections/summary-section.tsx`)
   - The "Default summary prompt" dropdown maps only `SUMMARY_PRESETS` — no custom prompts listed, no create UI.
   - **Data-loss bug:** line ~65 writes `customPrompts: []` on *every* preset change, wiping any stored custom prompts:
     ```ts
     body: JSON.stringify({ summaryPrompt: { selectedPrompt: value, customPrompts: [] } })
     ```

2. **Per-recording summary control** (`src/hooks/use-transcription-summary.ts`)
   - Line ~43 `useState("general")` — the dropdown always initializes to `general`, ignoring the saved `summaryPrompt.selectedPrompt`.
   - Line ~94 `body: JSON.stringify({ preset: summaryPreset })` — re-generate always sends an explicit `preset`, which overrides the saved default via the route's `presetId || promptConfig.selectedPrompt`.

3. **Per-recording dropdown options** (`src/components/recordings/transcription-section.tsx:194` and `src/components/dashboard/transcription-panel.tsx`)
   - Both map only `SUMMARY_PRESETS`, so a custom prompt can never be picked per recording.

Net effect: even after writing a valid custom prompt into `summaryPrompt` via the API, it is never selectable and never used, and touching the Settings → Summary dropdown deletes it.

## Proposed Solution

Mirror the existing **title** prompt manager (`src/components/settings-sections/prompt-manager.tsx`, which already does presets + custom CRUD + an active selector) for **summary** prompts:

1. `summary-section.tsx`: add create/edit/view/delete for custom summary prompts; list customs in the default dropdown; stop sending `customPrompts: []` (preserve on every save).
2. `use-transcription-summary.ts`: initialize `summaryPreset` from the saved `summaryPrompt.selectedPrompt` (fetched in the existing GET), and include custom prompts as selectable options.
3. `transcription-section.tsx` + `transcription-panel.tsx`: render custom prompts alongside presets in the per-recording dropdown.

I already have a working patch for (1) (typecheck/lint/test/build green locally) and am happy to open a PR covering all three surfaces — but since the feature spans several files and the per-recording UX is a design choice, I wanted to check first how maintainers would prefer it structured (e.g. one PR vs. settings-first, and whether the per-recording dropdown should list customs or just inherit the default).

## Alternatives Considered

- **Inject the custom prompt via the authenticated `PUT /api/settings/user`** (works today because the backend honors it) — but it's erased the moment you open Settings → Summary, and it's still never used by per-recording re-generate. Workaround only.
- Keep custom prompts title-only — but the summary is where a custom prompt matters most for self-hosters.

## Environment

- Riffado v0.5.3 (self-hosted, Docker, `ghcr.io/openplaud/openplaud`); behavior also verified in source on `main` @ `33f0270`.

### PR DESCRIPTION
**Pull request by kacperkwapisz**
## Description

Custom summary prompts were fully supported on the backend (`SummaryPromptConfiguration`, `getAllSummaryPrompts`, the resolve logic in `POST /api/recordings/[id]/summary`) but unreachable from the app: Settings → Summary only listed the four built-in presets, and every preset-change save overwrote `customPrompts` with an empty array, silently deleting any custom prompt on the next save. This PR wires up the missing UI and hardens the edge cases that come with it.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Dependency update

## Related Issues

Fixes #199

## Changes Made

- Settings → Summary now has full CRUD (create/edit/view/delete) for custom summary prompts, mirroring the existing title-prompt manager, and lists custom prompts alongside the built-in presets in the default-prompt dropdown.
- Fixed the data-loss bug: preset-change saves now round-trip the current `customPrompts` array instead of hardcoding `customPrompts: []`.
- The per-recording summary dropdown (recording detail page and dashboard) now lists custom prompts too, and initializes to the user's saved default instead of always defaulting to `"general"`.
- `PUT /api/settings/user` now validates the shape of `summaryPrompt` before encrypting/storing it, instead of accepting any JSON-serializable value.
- `POST /api/recordings/[id]/summary` reports the prompt id it actually used and whether it fell back to the default (e.g. the requested custom prompt was deleted in another session); the client now shows a warning toast instead of a misleading success toast in that case.
- The custom-prompt editor warns (non-blocking) when a prompt is missing the `{transcription}` placeholder.
- Added `CHANGELOG.md` entries and a regression test for the new validator and merge/resolve helpers.

## Testing

- `pnpm format-and-lint:fix`
- `pnpm type-check`
- `pnpm test` (853 passed, 10 skipped)

### Test Environment
- OS: Linux (dev container)
- Node version: per repo `.nvmrc`/`package.json`
- Browser (if UI changes): manual review of changed components, no live browser QA performed in this session

### Test Steps
1. Settings → Summary → create a custom summary prompt, set it as default, reload the page, confirm it's still listed and selected.
2. Change the default prompt preset, reload, confirm the custom prompt still exists (previously it was wiped).
3. Open a recording with a transcript, confirm the summary dropdown lists the custom prompt and the saved default is preselected; regenerate and confirm it's used.

## Checklist

- [x] My code follows the style guidelines of this project (`pnpm format-and-lint`)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (CHANGELOG)
- [x] My changes generate no new warnings or errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`pnpm test`)
- [x] Type checking passes (`pnpm type-check`)
- [ ] Any dependent changes have been merged and published

## Database Changes

- [ ] I have generated a new migration (`pnpm db:generate`)
- [ ] I have tested the migration locally
- [ ] I have included rollback instructions (if applicable)

No schema, env var, or `docker-compose.yml` changes — this is UI wiring and validation against the existing `userSettings.summaryPrompt` encrypted-JSON column.

## Breaking Changes

None.

## Additional Notes

Two commits: the feature itself (`feat(ai): expose custom summary prompts in the UI`), then a follow-up hardening pass (`fix(ai): harden custom summary prompt edge cases`) covering validation, the deleted-prompt fallback warning, and regression tests.

## For Reviewers

- `src/app/api/settings/user/route.ts`: shape validation for `summaryPrompt` — confirm the validator's field list matches `CustomSummaryPrompt`.
- `src/hooks/use-transcription-summary.ts`: the per-recording hook now fetches `/api/settings/user` once on mount to seed the default/options; worth checking this doesn't over-fetch given existing call sites.

---
## Summary by cubic
Expose custom summary prompts in Settings and recording views with full CRUD and safe defaults. Also surfaces prompt fallback info in both manual and auto-summarize flows.

- **New Features**
  - Settings → Summary: manage custom prompts (create, edit, view, delete) and set a default alongside presets.
  - Recording summary dropdowns list custom prompts and initialize to the saved default.
  - Editor warns when the `{transcription}` placeholder is missing.
  - Summary generation returns the used prompt id and fallback status from the shared generator; the client shows a warning toast on fallback (manual and auto-summarize).

- **Bug Fixes**
  - Preset changes no longer wipe `customPrompts`; the current list is round-tripped on save.
  - `PUT /api/settings/user` validates the `summaryPrompt` shape before storing.
  - Optimistic create/edit/delete of custom prompts now rolls back on failure to avoid local/server divergence.
  - Prevent settings fetch from overwriting a user’s manual prompt pick in recording views.

<sup>Written for commit 4cfe2b707fe51969b36502fdf0862d8254549531. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/riffado/riffado/pull/272?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/components/settings-sections/summary-section.tsx [hunk 1]
```diff
@@ -51,31 +76,93 @@ export function SummarySection() {
...
+
+        try {
+            await savePromptConfig({
+                selectedPrompt,
+                customPrompts: updatedPrompts,
+            });
+            toast.success("Prompt saved");
+        } catch {
+            toast.error("Failed to save prompt");
+        }
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/components/settings-sections/summary-section.tsx:179]: Confirmed and fixed in 4ed27c9 (`fix(ai): roll back custom prompt state on failed save/delete`).

You're right: `handleSaveCustomPrompt` and `handleDeleteCustomPrompt` applied `setCustomPrompts(updatedPrompts)` optimistically but never reverted it in the `catch` block on a failed `PUT /api/settings/user`. A rejected create/edit/delete stayed in local state, so a later unrelated save (e.g. `handlePresetChange`) would send that divergent array via `savePromptConfig` and persist the previously rejected mutation as if it had succeeded.

Both handlers now capture `previousPrompts` (and `previousSelectedPrompt` for delete, since it can also change `selectedPrompt`) before the optimistic update, and restore them in `catch`, matching the revert pattern `handlePresetChange`/`handleLanguageChange` already use elsewhere in this file.

File: src/hooks/use-transcription-summary.ts [hunk 2]
```diff
@@ -41,12 +60,49 @@ export function useTranscriptionSummary({
...
+            .then((res) => (res.ok ? res.json() : null))
+            .then((data) => {
+                const config = data?.summaryPrompt as
+                    | SummaryPromptConfiguration
+                    | null
+                    | undefined;
+                if (!config) return;
+                if (config.selectedPrompt) {
+                    setSummaryPreset(config.selectedPrompt);
+                }
```
[the remark this answers is not in the document]
Comment (kacperkwapisz) [reply src/hooks/use-transcription-summary.ts]: Confirmed and fixed in 596e2d0 (`fix(ai): don't let the settings fetch clobber a manual prompt pick`).

The settings-fetch effect fires once on mount, at the same time a user could plausibly interact with the per-recording dropdown. `setSummaryPreset` was just the raw `useState` setter passed straight through as `onValueChange`, so if the user picked a prompt before the `/api/settings/user` response arrived, the fetch's `.then()` would unconditionally overwrite it back to the saved default via `setSummaryPreset(config.selectedPrompt)` -- "Summarize" would then run with a prompt the user never chose.

Fixed by wrapping the setter: `setSummaryPreset` (the one returned to callers) now flips a `userSelectedPresetRef` to `true` the moment it's called explicitly, and the settings-fetch effect only applies the fetched default (via the internal `setSummaryPresetState`) when that ref is still `false`.
